src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDependencyDao.java

```diff
@@ -75,16 +75,17 @@ public boolean datasetHasSnapshotReference(Dataset dataset) throws InterruptedEx
     FireStoreProject fireStoreProject =
         FireStoreProject.get(dataset.getProjectResource().getGoogleProjectId());
     String dependencyCollectionName = getDatasetDependencyId(dataset.getId().toString());
-    CollectionReference depColl =
-        fireStoreProject.getFirestore().collection(dependencyCollectionName);
+    Firestore firestore = fireStoreProject.getFirestore();
+    CollectionReference depColl = firestore.collection(dependencyCollectionName);
 
     Query query = depColl.whereEqualTo("snapshotId", snapshotId);
     int batchSize = configurationService.getParameterValue(FIRESTORE_QUERY_BATCH_SIZE);
-    FireStoreBatchQueryIterator queryIterator = new FireStoreBatchQueryIterator(query, batchSize);
+    FireStoreBatchQueryIterator queryIterator =
+        new FireStoreBatchQueryIterator(query, batchSize, fireStoreUtils);
 
     List<String> fileIds = new ArrayList<>();
     for (List<QueryDocumentSnapshot> batch = queryIterator.getBatch();
-        batch != null;
+        batch != null && !batch.isEmpty();
```

Comment:
FWIW, I think you can combine this with:
```suggestion
        !CollectionUtils.isEmpty(batch)
```

Reply:
Oh perfect, thanks!